DataPlotly/core/plot_factory.py

```diff
@@ -65,12 +66,14 @@ class PlotFactory(QObject):  # pylint:disable=too-many-instance-attributes
 
     plot_built = pyqtSignal()
 
-    def __init__(self, settings: PlotSettings = None, visible_region: QgsReferencedRectangle = None):
+    def __init__(self, settings: PlotSettings = None, layout_item: QgsExpressionContextGenerator = None,
```

Comment:
instead of layout_item here this should be 'context_generator' -- for plots inside the canvas dock the context will be taken from the canvas, not a layout item